src/shared/connection/connection_poll.cpp

```diff
@@ -0,0 +1,114 @@
+#include "connection_poll.hpp"
+//#include "connection.hpp"
+#include "logger.hpp"
+
+#include <unistd.h>
+
+ConnectionPoll::ConnectionPoll(int t_timeout)
+    : m_timeout(t_timeout){
+
+  for (std::size_t i = m_poll_index; i < m_max; i++) { /* initialize fds to -1 */
+    m_poll[i].fd = -1;
+  }
+}
+
+bool ConnectionPoll::add_socket(int t_socket_fd) {
+
+  /* add connected sockets to polling system */
+
+  if (m_poll_index >= m_max) {
+    return false;
+  }
+
+  m_poll[m_poll_index].fd = t_socket_fd;
+  m_poll[m_poll_index].events = 0;
+  m_poll[m_poll_index].events |= POLLIN;
+  m_poll_index++;
+
+  return true;
+}
+
+bool ConnectionPoll::accept_awaits() {
+
+  if (poll(m_poll, 1, m_timeout) > 0) {
```

Comment:
We could add a define or a variable called `zero` and make it `=1` for less confusion

Reply:
Can we also make a variable and then check the return of bigger than 0

```
int res = poll(m_poll, 1, m_timeout);
if( res > 0){
}
```